synapseclient/client.py

```diff
@@ -1413,12 +1337,12 @@ def setAnnotations(self, entity, annotations={}, **kwargs):
 
         :returns: A dictionary
         """
-
-        uri = '/entity/%s/annotations' % id_of(entity)
+        entity = self.get(entity,downloadFile=False)
+        uri = '/entity/%s/annotations' % entity.id
```

Comment:
This looks like it's addressing SYNPY-362, which came about as a consequence of stripping out the redundant synapse properties from annotations (SYNPY-175).

I agree w/ Larsson. We might only look up the etag if we don't already have it (in the entity), possibly something like:

   if 'etag' not in synapseAnnos:
       if 'etag' in entity:
           synapseAnnos['etag'] = entity['etag']
       else:
           old_annos = syn.restGET(uri)
           synapseAnnos['etag'] = old_annos['etag']

Whether it's worth an extra couple lines, I don't know. In the common case, syn.store(entity) already adds the etag of the entity to its annotations before calling setAnnotations(). It's subtleties like this that'll make you crazy.

As an aside, some think setAnnotations should be changed to work only in memory as the R client does. See SYNPY-232.